src/ape/plugins/__init__.py

```diff
@@ -145,26 +153,33 @@ def get_plugin_name_and_hookfn(h):
             return h.plugin_name, getattr(h.plugin, attr_name)()
 
         for plugin_name, results in map(get_plugin_name_and_hookfn, hookimpls):
+
             # NOTE: Some plugins return a tuple and some return iterators
-            if not isinstance(results, tuple) and hasattr(results, "__iter__"):
+            if not isinstance(results, Generator):
```

Comment:
Other comments hinted at this - but wondering if we are not handling tuples correctly in here